log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -368,6 +419,27 @@ public void flush()
         flushMessageOutputSafe();
     }
 
+    /**
+     * Requests a full flush of all currently queued messages from the background thread. If the handler has already been closed,
+     * this method will return an empty {@link Optional}. Otherwise, the returned {@link Optional} will contain a {@link ListenableFuture}
+     * that will complete when the background flush has completed.
+     */
+    public Optional<ListenableFuture<Void>> requestFullFlush()
```

Comment:
Why use an optional here.  In the empty case just return `Futures.getDoneFuture(null)`

Reply:
The idea was that the caller could distinguish between a flush that was respected (and processed immediately) vs a flush that was ignored, but I guess the question would be "what are you gonna do when you realize your flush was ignored, log something?" - I've removed the `Optional` wrapping.